libpyvinyl/BaseData.py

```diff
@@ -0,0 +1,477 @@
+""" :module BaseData: Module hosts the BaseData class."""
+from typing import Union
+from abc import abstractmethod, ABCMeta
+from libpyvinyl.AbstractBaseClass import AbstractBaseClass
+
+
+class BaseData(AbstractBaseClass):
+    """The abstract data class. Inheriting classes represent simulation input and/or output
+    data and provide a harmonized user interface to simulation data of various kinds rather than a data format.
+    Their purpose is to provide a harmonized user interface to common data operations such as reading/writing from/to disk.
+
+    :param key: The key to identify the Data Object.
+    :type key: str
+    :param expected_data: A placeholder dict for expected data. The keys of this dict are expected to be found during the execution of `get_data()`.
+    The value for each key can be `None`.
+    :type expected_data: dict
+    :param data_dict: The dict to map by this DataClass. It has to be `None` if a file mapping was already set, defaults to None.
+    :type data_dict: dict, optional
+    :param filename: The filename of the file to map by this DataClass. It has to be `None` if a dict mapping was already set, defaults to None.
+    :type filename: str, optional
+    :param file_format_class: The FormatClass to map the file by this DataClass, It has to be `None` if a dict mapping was already set, defaults to None
+    :type file_format_class: class, optional
+    :param file_format_kwargs: The kwargs needed to map the file, defaults to None.
+    :type file_format_kwargs: dict, optional
+    """
+
+    def __init__(
+        self,
+        key: str,
+        expected_data: dict,
+        data_dict: dict = None,
```

Comment:
To type hint something that can be both `obj` and `None`, you can use `Optional[obj]`, which is equivalent to `Union[obj, None]`. If the `None` type is not marked, mypy and IDEs will probably complain about it.

Reply:
Done. Thanks for the hints.